src/common/ui/code_extras/mode/javascript-mixed/javascript-mixed.js

```diff
@@ -0,0 +1,304 @@
+// Javascript mixed mode for CodeMirror
+// Distributed under an MIT license
+
+/* eslint func-names: "off" */
+(function (mod) {
+  if (typeof exports === 'object' && typeof module === 'object') { // CommonJS
+    // eslint-disable-next-line global-require
+    mod(require('codemirror/lib/codemirror'), require('codemirror/mode/xml/xml'), require('codemirror/mode/javascript/javascript'), require('codemirror/mode/css/css'));
+  // eslint-disable-next-line no-undef
+  } else if (typeof define === 'function' && define.amd) { // AMD
+    // eslint-disable-next-line no-undef
+    define(['codemirror/lib/codemirror', 'codemirror/mode/xml/xml', 'codemirror/mode/javascript/javascript', 'codemirror/mode/css/css'], mod);
+  } else { // Plain browser env
+    // eslint-disable-next-line no-undef
+    mod(CodeMirror);
+  }
+}((CodeMirror) => {
+  function dbg() {
```

Comment:
FWIW you can use process.env.DEBUG, otherwise an empty function looks like an accidental leftover.

```js
  function dbg(...args) {
    // eslint-disable-next-line no-console
    if (process.env.DEBUG) console.debug(...args);
  }
```

On the other hand, maybe it's time to remove all the `dbg` calls to cleanup the code (in at least one case it'll eliminate the need for an intermediate variable).

Reply:
- how do I make the extension codes running on Firefox see `process.env.DEBUG`? Do I start firefox from command line and add the `DEBUG` to the environment?

- I'd feel more comfortable with removing the `dbg` calls after the logic has been tired out more thoroughly, say, in some beta. They are quite useful in tracing/debugging.

Reply:
AFAIK The usage depends on IDE or the way you run the build so refer to the respective documentation or try googling it.

Reply:
I added `process.env` check, though I don't really understand how nodejs's `process.env` got simulated in web extension. 

I also added a global check `window.DEBUG_CM_JAVASCRIPT_MIXED` for now, so that I can turn on /off from devtools.

Reply:
Run node.js with a command line that sets the DEBUG variable - it's usually possible to set via some config in an IDE, see the documentation. Don't add a `window` check because it preserves this code in in the production build whereas with just process.env check all dbg code and calls are eliminated.

Reply:
`Run node.js with a command line that sets the DEBUG variable`
But violentmoneky, being a web extension, does not run in `node.js`. The process.env it got is empty - the `process` is some npm implementation for a a browser that returns empty `env`.
I confirmed it by starting firefox from command line with `DEBUG` added to the envrionment. The `env` remains empty.

Reply:
You need to build the extension with this variable defined. The build process runs node.js internally but that's probably an unnecessary info so I don't know which IDE you are using but I suppose it should be easy to find a tutorial on setting up environment variables for any current IDE, as well as in case you build it using a command line.

Reply:
Ah, I see what you mean: webpack does the magic at build time.
The code is changed.